packages/react-native-tracer-provider/native-src/android/app/src/main/java/io/embrace/reactnativetracerprovider/ReactNativeTracerProviderModule.kt

```diff
@@ -0,0 +1,337 @@
+
+            when (type) {
+                ReadableType.Boolean -> builder.put(key, map.getBoolean(key))
+                ReadableType.Number -> builder.put(key, map.getDouble(key))
+                ReadableType.String -> builder.put(key, map.getString(key) ?: "")
+                ReadableType.Array -> {
+                    val array = map.getArray(key)
+                    val arrayType = array?.getType(0)
+                    when (arrayType) {
+                        ReadableType.Boolean -> builder.put(AttributeKey.booleanArrayKey(key), booleanListFromReadableArray(array))
+                        ReadableType.Number -> builder.put(AttributeKey.doubleArrayKey(key), doubleListFromReadableArray(array))
+                        ReadableType.String -> builder.put(AttributeKey.stringArrayKey(key), stringListFromReadableArray(array))
+                        else -> {
+                            log.warning("invalid attribute key: $key")
+                        }
+                    }
+                }
+                else -> {
+                    log.warning("invalid attribute value for key: $key")
+                }
+            }
+        }
+
+        return builder.build()
+    }
+
+    private fun spanContextFromReadableMap(map: ReadableMap): SpanContext {
+        val traceId = map.getString(SPAN_CONTEXT_TRACE_ID_KEY) ?: ""
+        val spanId = map.getString(SPAN_CONTEXT_SPAN_ID_KEY) ?: ""
+
+        val spanContext = SpanContext.create(
+            traceId,
+            spanId,
+            TraceFlags.getDefault(),
+            TraceState.getDefault()
+        )
+
+        return spanContext
+    }
+
+    private fun spanContextToWritableMap(spanContext: SpanContext?): WritableMap {
+        val map = this.writableMapBuilder.build()
+
+        map.putString(SPAN_CONTEXT_TRACE_ID_KEY, spanContext?.traceId ?: "")
+        map.putString(SPAN_CONTEXT_SPAN_ID_KEY, spanContext?.spanId ?: "")
+
+        return map
+    }
+
+    init {
+        writableMapBuilder = WritableNativeMapBuilder()
+    }
+
+    /**
+     * Exposed for unit testing to allow a different trace provider and writableMapBuilder to be injected
+     */
+    constructor(
+        reactContext: ReactApplicationContext,
+        tracerProvider: TracerProvider,
+        writableMapBuilder: WritableMapBuilder
+    ) : this(reactContext) {
+        this.tracerProvider = tracerProvider
+        this.writableMapBuilder = writableMapBuilder
+    }
+
+    private fun getTracerKey(name: String, version: String, schemaUrl: String): String {
+        return "$name $version $schemaUrl"
+    }
+
+    private fun getSpan(id: String): Span? {
+        val span = spans[id]
+        if (span == null) {
+            log.warning("could not retrieve span with id: $id")
+        }
+        return span
+    }
+
+    /**
+     * Methods to allow the JS side to conform to @opentelemetry-js/api
+     */
+
+    @ReactMethod fun getTracer(name: String, version: String, schemaUrl: String) {
```

Comment:
Annotations are usually placed above the function, like this:

```
@ReactMethod
fun getTracer
```

Reply:
updated